llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -1868,6 +1868,13 @@ Instruction *InstCombinerImpl::visitAdd(BinaryOperator &I) {
   if (Instruction *Res = foldBinOpOfSelectAndCastOfSelectCondition(I))
     return Res;
 
+  if (Changed) {
+    for (User *U : I.users()) {
+      if (auto *PHI = dyn_cast<PHINode>(U))
+        Worklist.pushUsersToWorkList(*PHI);
+    }
+  }
```

Comment:
This obviously has a positive effect, but I don't quite understand why this isn't already handled by `pushUsersToWorkList` in the normal InstCombine loop.

Reply:
> I don't quite understand why this isn't already handled by pushUsersToWorkList in the normal InstCombine loop

Take the above case as an example. If the add instruction is updated, `%iv` and `%phi` will be re-queued. But `%cmp` cannot be re-queued since we do not change `%iv`.

Reply:
Oh yeah duh